### PR TITLE
Add support for Tuples

### LINKED ISSUE #12: Add tuples
Tuples are of the form `{"foo", 123}` which is of the type: `{String, Int}`. They are like lists, but what they will contain is already fixed at compile time and cannot contain arbitrary values.

### PR DESCRIPTION
**Pull request by sreecodeslayer**
Closes #12

### CODE REVIEW

File: lib/fika/parser.ex [hunk 1]
```diff
@@ -121,6 +121,14 @@ defmodule Fika.Parser do
     |> ignore(string("]"))
     |> Helper.to_ast(:exp_list)
 
+  tuple =
+    ignore(string("{"))
+    |> concat(allow_space)
+    |> concat(list_content)
```
Comment (sreecodeslayer) [inline lib/fika/parser.ex:133]: @emilsoman should we change `list_content` since its the same for tuples as well? 
Maybe call it as `enum_content` ?

Comment (emilsoman) [reply lib/fika/parser.ex:133]: Tuple has a list-like content, so let's keep it as it is

File: README.md [hunk 2]
```diff
@@ -95,6 +95,9 @@ x = true
 # Type: List(Int)
 list_of_ints = [1, 2, 3]
 
+# Type: Tuple(String)
+tuple_of_ints = {"this", "is", "a", "tuple", "of", "strings!"}
```
Comment (polvalente) [inline README.md]: Maybe should rename the variable?

File: test/fika/parser_test.exs [hunk 3]
```diff
@@ -401,6 +401,50 @@ defmodule Fika.ParserTest do
...
+    test "parses tuple with one element" do
+      str = """
+      {1}
+      """
+
+      {:ok, result, _rest, _context, _line, _byte_offset} = Parser.expression(str)
+      assert result == [{:tuple, {1, 0, 3}, [{:integer, {1, 0, 2}, 1}]}]
+    end
+
+    test "parses tuple with multiple elements" do
```
Comment (polvalente) [inline test/fika/parser_test.exs:439]: Should there be a test that shows empty tuples aren't allowed?

File: test/fika/type_checker_test.exs [hunk 4]
```diff
@@ -167,6 +167,62 @@ defmodule Fika.TypeCheckerTest do
...
+
+      assert {:ok, "Tuple(Tuple(Int))", _} = TypeChecker.infer_exp(Env.init(), ast)
+    end
+
+    test "tuple of tuple of mixed types" do
+      str = ~s({{1, 2}, {"3", 4}})
+
+      ast = Fika.Parser.expression!(str)
+
+      assert {
```
Comment (polvalente) [inline test/fika/type_checker_test.exs]: Tagged tuples aren't allowed then?

edit: by "tagged tuples", I mean things like `{:ok, ...}` and `{:error, ...}`

Comment (sreecodeslayer) [reply test/fika/type_checker_test.exs]: Yea, what we want is to support tuple with mixed type. Just added them 😉

File: test/fika/parser_test.exs (hunk 3 above)
Comment (sreecodeslayer) [reply test/fika/parser_test.exs:439]: empty tuple is `{}`, but that's already inferred as empty record `{}` which by itself is not allowed 😄

Comment (polvalente) [reply test/fika/parser_test.exs:439]: Nice!

**Comment by sreecodeslayer** [comment]
oops i had auto format toggled on, let me revert those changes 😂

**Comment by emilsoman** [comment]
Thanks @sreecodeslayer ❤️

**Pull request merged**